File: Python/0001-two-sum.py
class Solution:

    def twoSum(self, nums, target):
        
        num_indices = {}  # Maps nums[i] to i
        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_indices:
                return [num_indices[complement], i]
            num_indices[num] = i
        return [-1, -1]

File: Java/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        if (nums.length > 2){

            HashMap <Integer, Integer> hmap = new HashMap<>();
            k
            for (int i = 0; i < nums.length; i++){
                if (!hmap.isEmpty() && hmap.containsKey(target - nums[i])){
                    return new int[] {hmap.get(target - nums[i]), i};
                } else hmap.put(nums[i], i);
            }
        }

        return new int[] {0,1};
    }
}

